### PR TITLE
Add FyneDesk variant

### LINKED ISSUE #216: (Idea) FyneDesk edition
FyneDesk is a BSD licensed desktop written in Go. And it runs on FreeBSD:

image

Source: https://github.com/fyne-io/fynedesk/issues/132

>> In a discussion over at #freebsd on freenode the question came up whether it would be possible to build a GNU/GPL-free desktop using FyneDesk. Would it? Or are there some hard dependencies on anything GNU/GPL?

@Fyne_io:

> Yes that should be possible :) Everything in the Fyne project is BSD licensed. In terms of GNU that is harder to answer definitively but FyneDesk is modular so we can switch in/out sections if required.

Source: https://twitter.com/Fyne_io/status/1312848007629594628?s=20

### PR DESCRIPTION
**Pull request by probonopd**
FyneDesk is a BSD licensed desktop written in Go.
Closes #216.

### CODE REVIEW

**Comment by andydotxyz** [comment]
Currently the backlight control uses xbacklight executable so if that was installed it might be good.

**Comment by pkgdemon** [comment]
@probonopd can you tell me if these are errors?

```
Number of packages to be installed: 1
[1/1] Installing furybsd-fynedesk-live-settings-0...
Extracting furybsd-fynedesk-live-settings-0: 100%
hostname:  -> furybsd
slim_enable:  -> YES
Updating FreeBSD repository catalogue...
FreeBSD repository is up to date.
All repositories are up to date.
Checking integrity... done (0 conflicting)
The most recent versions of packages are already installed
Already on 'develop'
Your branch is up to date with 'origin/develop'.
# github.com/go-gl/glfw/v3.3/glfw
In file included from /root/go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20200625191551-73d3c3675aa3/native_linbsd.go:10:
In file included from /root/go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/include/GLFW/glfw3native.h:114:
In file included from /usr/local/include/GL/glx.h:32:
/usr/local/include/GL/gl.h:61:11: warning: 'GLAPIENTRY' macro redefined [-Wmacro-redefined]
/root/go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20200625191551-73d3c3675aa3/glfw/include/GLFW/glfw3.h:5863:10: note: previous definition is here
/tmp
```
I am getting this when running locally twice in a row.  I just want to make sure this is also designed to build, and rebuild locally not just with CI before merging.

**Comment by pkgdemon** [comment]
@probonopd Also FYI the ISO has artifact to continuous tag even though this is a pull request.  Is that expected?  For some reason I thought you made a patch so the ISO would not be artifacted to the continuous tag until the PR was merged into master?  https://github.com/furybsd/furybsd-livecd/releases/tag/continuous

**Comment by andydotxyz** [comment]
Oh, can you change the slim login image as well? 
If you copy the fynedesk file from theme/lochfyne.jpg to /usr/local/share/slim/themes/default/background.jpg it should look like this:

Screenshot 2020-10-09 at 23 08 28

**Comment by probonopd** [comment]
> @probonopd can you tell me if these are errors?

Accroding to @andydotxyz those can be ignored. Would still be nice to see them go away.

**Comment by probonopd** [comment]
> @probonopd Also FYI the ISO has artifact to continuous tag even though this is a pull request. Is that expected? For some reason I thought you made a patch so the ISO would not be artifacted to the continuous tag until the PR was merged into master?

Only branches within this repo get uploaded. PRs from outside this repo would not get uploaded.

Well. The whole "upload to GitHub Releases" thing is working way differently from how it should be, thanks to Cirrus CI cancelling builds whenever tags are created. I hope they will change this, which would allow us to attach artifacts to tags that we could create on-the-fly. Another way would be not to upload PRs to GitHub Releases at all, but in that case we'd need to put them somewhere else.

Let me write an issue for Cirrus CI.

**Comment by probonopd** [comment]
* Boot the Live ISO on an Intel-GPU based machine
* In xterm, enter: `sudo furybsd-xorg-tool`, select Intel, select the first of the options, restart Xorg
* Log in through slim (Username: `liveuser`, no password)
* There is a full-screen 1080p desktop. On top of it, there is an "Embedded Fyne Desktop" window
* When closing this window, Xorg gets killed

Any hints appreciated @andydotxyz

**Comment by probonopd** [comment]
Consider adding

* `freebsd-install` (GUI) or `furybsd-install` (CLI)
* `furybsd-xorg-tool`
* `furybsd-wifi-tool`
* Tool to set the keyboard layout

But currently these things are not written in Fyne, so they pull in Gtk/Qt...

**Comment by probonopd** [comment]
I have added them to the Dock, but when they are invoked their windows do not seem to show up. Why? Is this wrong?

https://github.com/furybsd/furybsd-livecd/blob/a97c42684ddaf0d955f22726b0ed4663460a2be7/overlays/uzip/furybsd-fynedesk-live-settings/files/usr/share/skel/dot.config/fyne/io.fyne.fynedesk/preferences.json

File: overlays/uzip/furybsd-fynedesk-live-settings/files/usr/local/bin/start-fynedesk [hunk 1]
```diff
@@ -3,6 +3,12 @@
 # Drop shadows
 compton &
 
+FYNE_FONT="Luxi Sans"
+export FYNE_FONT
+
+FYNE_THEME="light"
```
Comment (andydotxyz) [inline overlays/uzip/furybsd-fynedesk-live-settings/files/usr/local/bin/start-fynedesk:9]: environment variables override the system configuration.
This line will make the "Fyne settings" panel of the settings unable to change the theme.
If you want to force the user default to light consider using the .config/fyne/settings.json file instead.

**Comment by andydotxyz** [comment]
I think that the "fyne" tool can be helpful in setting up the environment as it can handle icons as well. 
I have just added BSD support to it so you'll need it from develop instead of the simple 'go get' line.

```
go get -d fyne.io/fyne/cmd/fyne
cd ~/go/src/fyne.io/fynd/cmd/fyne
git checkout 4ef15dd
go install
```

That will install the tool, we can then use it like this:
```
go get -d github.com/fyne-io/examples/
cd ~/go/src/github.com/fyne-io/examples/cmd/calculator
fyne install -icon ../../img/icons/calculator.png
```

This will build and install the app along with it's icon into the system.
I am working on a a new `fyne get` command that will automate even more of this :)

**Comment by probonopd** [comment]
> This will build and install the app along with it's icon into the system.

Can it use a custom `$PREFIX`? E.g., we don't want to install it "into the system" locally, but we want to install it to `${uzip}/usr/local`...

**Comment by andydotxyz** [comment]
Oh of course! silly me.
In this case I would use `fyne package` which will create a tarball. You can then expand that tarball to the correct place, or you could use the Makefile inside the tarball with PREFIX set.

**Comment by probonopd** [comment]
The wallpaper on the desktop is not as sharp as the same wallpaper in slim. Is this intentional?

**Comment by andydotxyz** [comment]
> The wallpaper on the desktop is not as sharp as the same wallpaper in slim. Is this intentional?

You’d need to look at the resolution of the screen vs the file and see. Probably different scaling algorithms. Is Slim also scaling or is it using pixels from the top-left?

**Comment by andydotxyz** [comment]
> Killing Xorg still results in an Embedded Fyne Desktop window

I guess the issue I opened a few hours ago still isn’t fixed? :-p https://github.com/fyne-io/fynedesk/issues/137

**Comment by probonopd** [comment]
`installer log` says: `/usr/bin/sysrc: no such file or directory` --> slim not working after installation.

Do the other editions also suffer from this?

**Comment by andydotxyz** [comment]
The latest build is looking really great, thanks @probonopd.

One more thought - can we add `arandr` to provide the GUI for screen management until FyneDesk provides it's own?

**Comment by andydotxyz** [comment]
Oh also, the username is set as 'liveuser' in the login screen after install. Can we set it blank instead please?

**Comment by probonopd** [comment]
> One more thought - can we add arandr to provide the GUI for screen management until FyneDesk provides it's own?

Added.

> Oh also, the username is set as 'liveuser' in the login screen after install. Can we set it blank instead please?

This is a known issue across the board, not specific to the FyneDesk variant. I think we should solve this in a generic way, not in this PR.

**Pull request closed without merging**